plugins/rbac-backend/src/service/policies-rest-api.ts

```diff
@@ -460,17 +463,37 @@ export class PolicesServer {
         }
       }
 
-      // enforcer.updateGroupingPolicy(oldRole, newRole) was not implemented
-      // for ORMTypeAdapter.
-      // So, let's compensate this combination delete + create.
-      const isRemoved = await this.enforcer.removeGroupingPolicies(oldRole);
-      if (!isRemoved) {
-        throw new Error('Unexpected error'); // 500
+      const metadata = await this.roleMetadata.findRoleMetadata(roleEntityRef);
+      if (!metadata) {
+        throw new NotFoundError(`Unable to find metadata for ${roleEntityRef}`);
+      }
+      if (metadata.source === 'csv-file') {
+        throw new Error(
+          `Role ${roleEntityRef} can be modified only using csv policy file.`,
+        );
+      }
+      if (metadata.source === 'default') {
+        throw new Error(
+          `Role ${roleEntityRef} can be modified only using application config`,
+        );
```

Comment:
The `EnforcerDelegate` has a `checkIfPolicyModifiable`, could we reuse / expand upon that for here?

Reply:
We considered to handle this part in the frame of another issue, related to improvement validations.